cpp/log/tree_signer-inl.h

```diff
@@ -100,44 +77,24 @@ uint64_t TreeSigner<Logged>::LastUpdateTime() const {
 
 template <class Logged>
 util::Status TreeSigner<Logged>::HandlePreviouslySequencedEntries(
+    const ct::SequenceMapping& sequence_mapping,
     std::vector<cert_trans::EntryHandle<Logged>>* pending_entries) const {
   CHECK(std::is_sorted(pending_entries->begin(), pending_entries->end(),
                        PendingEntriesOrder<Logged>()));
+  std::unordered_map<std::string, int64_t> sequenced_hashes;
+  for (auto& m : sequence_mapping.mapping()) {
+    CHECK(sequenced_hashes.insert(make_pair(m.unsequenced_hash(),
+                                            m.sequence_number())).second);
+  }
   // Check and handle any previously sequenced entries
   auto it(pending_entries->begin());
   while (it != pending_entries->end()) {
-    if (!it->Entry().has_provisional_sequence_number()) {
-      // There can't be any more entries with provisional sequence numbers now
-      // (because of the sort order) so we can early out.
-      break;
-    }
-    cert_trans::EntryHandle<Logged> presequenced;
-    util::Status status(consistent_store_->GetSequencedEntry(
-        it->Entry().provisional_sequence_number(), &presequenced));
-    if (status.ok()) {
-      CHECK(it->Entry().entry() == presequenced.Entry().entry() &&
-            it->Entry().sct() == presequenced.Entry().sct())
-          << "Pending entry with provisional_sequence_number:\n"
-          << it->Entry().DebugString() << "\n"
-          << "does not match sequenced entry for that sequence "
-          << "number:\n" << presequenced.Entry().DebugString()
-          << "\nclearing provisional_sequence_number to "
-          << "resequence pending entry";
-      // This entry is already sequenced just fine, no need to return it.
-      VLOG(1) << "Entry already sequenced: "
-              << util::ToBase64(it->Entry().Hash()) << " ("
-              << presequenced.Entry().sequence_number()
-              << "), removing from list";
+    if (sequenced_hashes.find(it->Entry().Hash()) != sequenced_hashes.end()) {
+      // This hash is already sequenced.
```

Comment:
This is actually dead code now, I've removed it.
But we couldn't have done that anyway because the entry in etcd doesn't get updated with the sequence number, that info lives in the `sequence_mapping` file.